source/engine/assets/manager/SkeletalAnimationAssetManager.cpp

```diff
@@ -0,0 +1,130 @@
+#include "SkeletalAnimationAssetManager.h"
+#include "AssetUtilities.h"
+#include "asset/AssetData.h"
+
+#include "ncasset/Import.h"
+
+#include <algorithm>
+
+namespace nc
+{
+SkeletalAnimationAssetManager::SkeletalAnimationAssetManager(const std::string& skeletalAnimationAssetDirectory, uint32_t maxSkeletalAnimations)
+    : m_assetDirectory{skeletalAnimationAssetDirectory},
+      m_maxSkeletalAnimationCount{maxSkeletalAnimations},
+      m_assetIds{},
+      m_skeletalAnimations{},
+      m_onUpdate{}
+{
+}
+
+bool SkeletalAnimationAssetManager::Load(const std::string& path, bool isExternal, asset_flags_type)
+{
+    if (m_assetIds.size() + 1 >= m_maxSkeletalAnimationCount)
+    {
+        throw NcError("Cannot exceed max skeletal animations count.");
+    }
+
+    if (IsLoaded(path))
+    {
+        return false;
+    }
+
+    const auto fullPath = isExternal ? path : m_assetDirectory + path;
+    m_assetIds.emplace_back(path);
+    m_skeletalAnimations.emplace_back(asset::ImportSkeletalAnimation(fullPath));
+
+    m_onUpdate.Emit(asset::SkeletalAnimationUpdateEventData{
+        std::span<const std::string>{m_assetIds},
+        std::span<const asset::SkeletalAnimation>{m_skeletalAnimations},
+        asset::UpdateAction::Load
+    });
+    return true;
+}
+
+bool SkeletalAnimationAssetManager::Load(std::span<const std::string> paths, bool isExternal, asset_flags_type)
+{
+    if (m_assetIds.size() + paths.size() >= m_maxSkeletalAnimationCount)
+    {
+        throw NcError("Cannot exceed max skeletal animations count.");
+    }
+
+    bool anyLoaded = false;
+
+    for(const auto& path : paths)
+    {
+        if (IsLoaded(path))
+        {
+            continue;
+        }
+
+        const auto fullPath = isExternal ? path : m_assetDirectory + path;
+        m_assetIds.emplace_back(path);
+        m_skeletalAnimations.emplace_back(asset::ImportSkeletalAnimation(fullPath));
+        anyLoaded = true;
+    }
+
+    if (anyLoaded)
+    {
+        m_onUpdate.Emit(asset::SkeletalAnimationUpdateEventData{
+            std::span<const std::string>{m_assetIds},
+            std::span<const asset::SkeletalAnimation>{m_skeletalAnimations},
+            asset::UpdateAction::Load
+        });
+    }
+    return anyLoaded;
+}
+
+bool SkeletalAnimationAssetManager::Unload(const std::string& path, asset_flags_type)
+{
+    auto pos = std::find(m_assetIds.begin(), m_assetIds.end(), path);
```

Comment:
Prefer ranges here so you don't have to name two iterators. Its slightly less error prone (not that this is wrong).